src/compiler/utilities.ts

```diff
@@ -3917,7 +3918,7 @@ namespace ts {
         if (c.charCodeAt(0) === CharacterCodes.nullCharacter) {
             return "&#0;";
         }
-        return jsxEscapedCharsMap.get(c) || encodeJsxCharacterEntity(c.charCodeAt(0));
+        return jsxEscapedCharsMap.get(c) ?? encodeJsxCharacterEntity(c.charCodeAt(0));

```

Comment:
`||` should have been fine, since none of the entries are the empty string.
```suggestion
        return jsxEscapedCharsMap.get(c) || encodeJsxCharacterEntity(c.charCodeAt(0));
```

Reply:
(same as above, and changed back too.)